jdaviz/tests/test_subsets.py

```diff
@@ -857,3 +858,93 @@ def test_delete_subsets(cubeviz_helper, spectral_cube_wcs):
     dc.remove_subset_group(dc.subset_groups[0])
 
     assert flux_viewer.toolbar.active_tool is None
+
+
+class TestRegionsFromSubsets:
+    """Tests for obtaining Sky Regions from subsets."""
+
+    def test_get_regions_from_subsets_cubeviz(self, cubeviz_helper, spectral_cube_wcs):
+
+        """ Basic tests for retrieving Sky Regions from spatial subsets in Cubeviz.
+        """
+        data = Spectrum1D(flux=np.ones((128, 128, 256)) * u.nJy, wcs=spectral_cube_wcs)
+        cubeviz_helper.load_data(data)
+
+        # basic test, a single circular region
+        cubeviz_helper.app.get_viewer('flux-viewer').apply_roi(CircularROI(25, 25, 10))
+        subsets = cubeviz_helper.app.get_subsets(include_sky_region=True)
+        sky_region = subsets['Subset 1'][0]['sky_region']
+        assert isinstance(sky_region, CircleSkyRegion)
+        assert_allclose(sky_region.center.ra.deg, 24.40786313)
+        assert_allclose(sky_region.center.dec.deg, 22.45185308)
+        assert_allclose(sky_region.radius.arcsec, 28001.08106569353)
+
+        # and that it is None when not specified
+        subsets = cubeviz_helper.app.get_subsets()
+        assert subsets['Subset 1'][0]['sky_region'] is None
+
+        # now test a composite subset, each component should have a sky region
+        cubeviz_helper.app.session.edit_subset_mode.mode = AndMode
+        cubeviz_helper.app.get_viewer('flux-viewer').apply_roi(CircularROI(30, 30, 10))
+
+        subsets = cubeviz_helper.app.get_subsets(include_sky_region=True)
+        assert len(subsets['Subset 1']) == 2
+        sky_region_0 = subsets['Subset 1'][0]['sky_region']
+        sky_region_1 = subsets['Subset 1'][1]['sky_region']
+        assert_allclose(sky_region_0.center.ra.deg, 24.40786313)
+        assert_allclose(sky_region_0.center.dec.deg, 22.45185308)
+        assert_allclose(sky_region_0.radius.arcsec, 28001.08106569353)
+        assert_allclose(sky_region_1.center.ra.deg, 28.41569583)
+        assert_allclose(sky_region_1.center.dec.deg, 25.44814949)
+        assert_allclose(sky_region_1.radius.arcsec, 25816.498273)
+
+        # and that they are both None when not specified
+        subsets = cubeviz_helper.app.get_subsets()
+        assert subsets['Subset 1'][0]['sky_region'] is None
+        assert subsets['Subset 1'][1]['sky_region'] is None
+
+    def test_get_regions_from_subsets_imviz(self, imviz_helper, spectral_cube_wcs):
```

Comment:
Shouldn't Imviz stick to 2D?

https://github.com/spacetelescope/jdaviz/blob/29370388be424fe1801d573b483d308b82c1ed48/jdaviz/conftest.py#L53

```suggestion
    def test_get_regions_from_subsets_imviz(self, imviz_helper, imviz_2d_wcs):
```

Reply:
yes, i am just accessing the spatial component of that wcs fixture